fdbclient/MultiVersionTransaction.actor.cpp

```diff
@@ -1228,11 +1211,12 @@ void MultiVersionApi::addNetworkThreadCompletionHook(void (*hook)(void*), void *
 	}
 	lock.leave();
 
-	auto hookPair = std::pair<void (*)(void*), void*>(hook, hook_parameter);
-	threadCompletionHooks.push_back(hookPair);
+	localClient->api->addNetworkThreadCompletionHook(hook, hook_parameter);
 
 	if(!bypassMultiClientApi) {
+		auto hookPair = std::pair<void (*)(void*), void*>(hook, hook_parameter);
 		for( auto it : externalClients ) {
+			MutexHolder holder(lock);
```

Comment:
Should this go outside of the loop to avoid creating a mutex holder in each iteration? I suppose if we expect the mode number of external clients to be zero and the next most common number to be one, then the answer is "no", it should stay in the loop.

Reply:
Those in theory should be the most common cases, but I'm not opposed to moving it out, either.

Reply:
This is no longer relevant with my most recent change.